content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/Methods.class/__methods__/checkmigration.rb

```diff
@@ -2,21 +2,55 @@
 # Description: This method checks to see if the VM has been migrated
 #
 
-# Get current provisioning status
-result = $evm.root["vm_migrate_task"].statemachine_task_status
-
-$evm.log('info', "returned <#{result}>")
-
-case result
-when 'error'
-  $evm.root['ae_result'] = 'error'
-  reason = $evm.root['miq_provision'].message
-  reason = reason[7..-1] if reason[0..6] == 'Error: '
-  $evm.root['ae_reason'] = reason
-when 'retry'
-  $evm.root['ae_result']      = 'retry'
-  $evm.root['ae_retry_interval'] = '1.minute'
-when 'ok'
-  # Bump State
-  $evm.root['ae_result'] = 'ok'
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Migrate
+          module StateMachines
+            class Checkmigration
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                checkmigration(task)
+              end
+
+              private
+
+              def task
+                raise "ERROR - task object not passed in" unless @handle.root['vm_migrate_task']
+                @handle.root['vm_migrate_task']
+              end
+
+              def checkmigration(task)
+                task_status = task['status']
+                result = task.statemachine_task_status
+
+                @handle.log('info', "CheckMigration returned <#{result}> for state <#{task.state}> and status <#{task_status}>")
+
+                case result
+                when 'error'
+                  @handle.root['ae_result'] = 'error'
+                  reason = @handle.root['vm_migrate_task'].message
+                  reason = reason[7..-1] if reason[0..6] == 'Error: '
+                  @handle.root['ae_reason'] = reason
+                when 'retry'
+                  @handle.root['ae_result'] = 'retry'
+                when 'ok'
+                  # Bump State
```

Comment:
@billfitzgerald0120 
Can we remove this comment, because we are not changing the state. When the method ends in 'ok' the engine automatically moves to the next state.